paasta_tools/paasta_remote_run.py

```diff
@@ -0,0 +1,135 @@
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import PaastaNotConfiguredError
+from paasta_tools.utils import SystemPaastaConfig
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import validate_service_instance
+
+from paasta_tools.native_mesos_scheduler import create_driver_with
+from paasta_tools.native_mesos_scheduler import compose_job_id
+from paasta_tools.native_mesos_scheduler import load_paasta_native_job_config
+
+from paasta_tools.frameworks.adhoc_scheduler import PaastaAdhocScheduler
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(description='')
+    add_remote_run_args(parser)
+    return parser.parse_args(argv)
+
+
+class UnknownPaastaRemoteServiceError(Exception):
+    pass
+
+
+def make_config_reader(instance_type):
+    def reader(service, instance, cluster, soa_dir=DEFAULT_SOA_DIR):
+        conf_file = '%s-%s' % (instance_type, cluster)
+        full_path = '%s/%s/%s.yaml' % (soa_dir, service, conf_file)
+        paasta_print("Reading paasta-remote configuration file: %s" % full_path)
+        config = service_configuration_lib.read_extra_service_information(service, conf_file, soa_dir=soa_dir)
+
+        if instance not in config:
+            raise UnknownPaastaRemoteServiceError(
+                'No job named "%s" in config file %s: \n%s' % (instance, full_path, open(full_path).read())
+            )
+
+        return config
+
+    return reader
+
+
+def main(argv=None):
+    args = parse_args(argv)
+    try:
+        system_paasta_config = load_system_paasta_config()
+    except PaastaNotConfiguredError:
+        paasta_print(
+            PaastaColors.yellow(
+                "Warning: Couldn't load config files from '/etc/paasta'. This indicates"
+                "PaaSTA is not configured locally on this host, and remote-run may not behave"
+                "the same way it would behave on a server configured for PaaSTA."
+            ),
+            sep='\n',
+        )
+        system_paasta_config = SystemPaastaConfig({"volumes": []}, '/etc/paasta')
+
+    service = figure_out_service_name(args, soa_dir=args.yelpsoa_config_root)
+    cluster = args.cluster or system_paasta_config.get_local_run_config().get('default_cluster', None)
+
+    if not cluster:
+        paasta_print(
+            PaastaColors.red(
+                "PaaSTA on this machine has not been configured with a default cluster."
+                "Please pass one using '-c'."),
+            sep='\n',
+            file=sys.stderr,
+        )
+        return 1
+
+    soa_dir = args.yelpsoa_config_root
+    dry_run = args.dry_run
+    instance = args.instance
+    command = args.cmd
+
+    if instance is None:
+        instance_type = 'adhoc'
+        instance = 'remote'
+    else:
+        instance_type = validate_service_instance(service, instance, cluster, soa_dir)
+
+    paasta_print('Scheduling a task on Mesos')
+    service_config = load_paasta_native_job_config(
```

Comment:
I don't quite understand, are we expecting these jobs to be "adhoc" jobs as in "adhoc-norcal-devc.yaml" ?

Because this looks like it *only* loads something from that class. 

I do see how this is convenient from a scheduler's perspective, but in the long term people are going to want to be able to run their existing "adhoc" and "chronos" jobs.